Docs: revamp index for AI/ML practitioners (#8523)
treeverse/lakeFS: docs/index.md

## Key lakeFS Features

* It is format-agnostic and works with both structured and unstructured data
* It works with numerous data tools and platforms.
* Your data stays in place, with no need to copy existing data
* It eliminates the need for data duplication using [zero-copy branching](understand/model.md#zero-copy-branching).
* It maintains high performance over data lakes of any size
* It includes configurable garbage collection capabilities
* It is proven in production and has an active community

<img src="{{ site.baseurl }}/assets/img/lakeFS_integration.png" alt="lakeFS integration into data lake" width="60%" height="60%" />

## How Does lakeFS Work With Other Tools? 

> lakeFS is an open source project that supports managing data in AWS S3, Azure Blob Storage, Google Cloud Storage (GCS), S3-Compatible storage solutins and even locally mounted directories. 

Reviewer: ```suggestion
lakeFS is an open source project that supports managing data in AWS S3, Azure Blob Storage, Google Cloud Storage (GCS), S3-Compatible storage solutions and even locally mounted directories. 
```

{: .pb-5 }

> With lakeFS, you can use any of the native integrations to read and write data directly from a repository, e.g.

Reviewer: ```suggestion
With lakeFS, you can use any of the tools and libraries you are used to work with to read and write data directly from a repository, e.g.
```

>>>
>>> df = pd.read_csv('lakefs://example-repository/main-branch/path/to.csv')
```

Additionally, lakeFS maintains compatibility with the S3 API to minimize adoption
friction. You can use it as a drop-in replacement for S3 from the perspective of
any tool interacting with a data lake.

For example, take the common operation of reading a collection of data from an object storage into a Spark DataFrame. For data outside a lakeFS repo, the code will look like this:

```py
>>> import boto3
>>> 
>>> s3 = boto3.resource('s3')
>>> obj = s3.Object('example-repository', 'main-branch/path/image.png')
>>> image_data = obj.get()['Body'].read()
```

You can use the same methods and syntax you are already using to read and write data when using a lakeFS repository. 
This simplifies the adoption of lakeFS - minimal changes are needed to get started, making further changes an incremental process.

## lakeFS is Git for Data

Git became ubiquitous when it comes to code because it had best supported engineering best practices required by developers, in particular:

* Collaborate during development.
* Reproduce and troubleshoot issues with a given version of the code
* Develop and Test in isolation

> * Revert code to a sable version in case of an error

Reviewer: ```suggestion
* Revert code to a stable version in case of an error
```

2. Troubleshooting production problems

    In some cases, a user might report inconsistencies, question the accuracy, or simply report data or inference results as incorrect. 
    
    Since data continuously changes, it is challenging to understand its state at the time of the error.

    With lakeFS you can create a branch from a commit to debug an issue in isolation.

👉🏻 [Read More](./understand/use_cases/reproducibility.html)

### Collaboration during development and training

With lakeFS, each member of the team can create their own branch, isolated from other people's changes. 

This allows to iterate on changes to an algorithm or transformation, without stepping on eachother's toes. These branches are centralized - they could be share among users for collaboration, and can even be merged.

With lakeFS you can even open [pull requests](./howto/pull-requests.md), allowing you to easily share changes with other members and collaborate on them.

### Isolated Dev/Test Environments with zero-copy branching

> lakeFS makes creating isolated dev/test environments for transformations, model development and ETL processes- achieved through the use of zero-copy branches.

Reviewer: ```suggestion
lakeFS makes creating isolated dev/test environments for transformations, model development, parallel experiments, and ETL processes- achieved through the use of zero-copy branches.
```

2. Troubleshooting production problems

    In some cases, a user might report inconsistencies, question the accuracy, or simply report data or inference results as incorrect. 
    
    Since data continuously changes, it is challenging to understand its state at the time of the error.

    With lakeFS you can create a branch from a commit to debug an issue in isolation.

👉🏻 [Read More](./understand/use_cases/reproducibility.html)

### Collaboration during development and training

With lakeFS, each member of the team can create their own branch, isolated from other people's changes. 

> This allows to iterate on changes to an algorithm or transformation, without stepping on eachother's toes. These branches are centralized - they could be share among users for collaboration, and can even be merged.

Reviewer 2: maybe
```suggestion
This allows to iterate on changes to an algorithm or transformation, without stepping on eachother's toes. These branches are centralized but also available locally - they could be share among users for collaboration, and can even be merged.
```
